### ISSUE TITLE
[Bug]: only some filetype @thing.filetype highlights working

### ISSUE THREAD

**Issue by mmirus** [olimorris/onedarkpro.nvim#101]
### Your OneDarkPro config

```lua
-- ignore default config and plugins
vim.opt.runtimepath:remove(vim.fn.expand("~/.config/nvim"))
vim.opt.packpath:remove(vim.fn.expand("~/.local/share/nvim/site"))

-- append test directory
local test_dir = "/tmp/onedarkpro"
vim.opt.runtimepath:append(vim.fn.expand(test_dir))
vim.opt.packpath:append(vim.fn.expand(test_dir))

-- install packer
local install_path = test_dir .. "/pack/packer/start/packer.nvim"
local install_plugins = false

if vim.fn.empty(vim.fn.glob(install_path)) > 0 then
    vim.cmd("!git clone https://github.com/wbthomason/packer.nvim " .. install_path)
    vim.cmd("packadd packer.nvim")
    install_plugins = true
end

local packer = require("packer")

packer.init({
    package_root = test_dir .. "/pack",
    compile_path = test_dir .. "/plugin/packer_compiled.lua",
})

-- install plugins
packer.startup(function(use)
    use("wbthomason/packer.nvim")
    use({ "nvim-treesitter/nvim-treesitter", run = ":TSUpdate" })
    use("olimorris/onedarkpro.nvim")

    -- Add any additional plugins here

    if install_plugins then
        packer.sync()
    end
end)

-- setup treesitter
local ok, treesitter = pcall(require, "nvim-treesitter.configs")
if ok then
    treesitter.setup({
        ensure_installed = "all",
        ignore_install = { "phpdoc" }, -- list of parser which cause issues or crashes
        highlight = { enable = true },
    })
end

-- setup onedarkpro
local ok, onedarkpro = pcall(require, "onedarkpro")
if ok then
    onedarkpro.setup({
      -- Your onedarkpro config here
    })
    vim.cmd("colorscheme onedarkpro")
end
```

### Error messages

N/A

### Expected

All filetype highlights should work. You should see something like this (using the deprecated `ft_highlights` along with the new filetype highlights:

![image](https://user-images.githubusercontent.com/524994/194096685-3b99cce0-c965-4fb6-a6d4-ded0fa8cb522.png)

### Actual

Only some work.

![image](https://user-images.githubusercontent.com/524994/194096521-c9a34c13-8d93-4144-8e67-7c7efa3277b7.png)

### Reproduce the bug

Take this test markdown file:

````md
# Working with @thing.markdown

["@text.title.markdown"] = { fg = theme.palette.red, style = config.options.bold },

## Here's a title

-----

["@punctuation.special.markdown"] = { fg = theme.palette.red },

1. Here's some special punctuation (the list item number)






# Not Working with @thing.markdown

["@text.literal.markdown"] = { fg = theme.palette.green },

Here is `inline code that should be green`.

-----

["@text.reference.markdown"] = { fg = theme.palette.blue },

This is for the name part of a link: [I should be blue](https://test.com).

-----

["@parameter.markdown"] = { fg = theme.palette.fg },

```bash
command these parameters should use fg
```

-----

["@punctuation.delimiter.markdown"] = { fg = theme.palette.gray },

The brackets and parentheses in the Markdown link syntax should be gray: [a link](https://test.com).

-----

["@uri.markdown"] = { fg = theme.palette.gray },

The URL in the Markdown link should be gray: [a link](https://test.com).
````

and open it in neovim with onedark and the new filetype highlights active.

You will see that most of the highlights don't apply.

-----

**Note:** I also experienced this problem _before_ the new filetype feature rolled out, as mentioned in #100:

> Note: using a non-ft_highlight like ["@text.title.markdown"] does not have this problem. Unfortunately, it seems like some highlight groups may not be available using that syntax, so I can't move fully over yet. (Or perhaps I'm doing something wrong.) Example: the ft_highlight TSLiteral = { fg = "${green}" } works (when caching is disabled) for styling inline code blocks, but the regular highlight ["@text.literal.markdown"] = { fg = "${green}" } does not.

In other words, it's not new to this latest version.

### Final checks

- [X] I have made sure that the colorscheme's caching is not the issue
- [X] I have made sure this is not a duplicate issue
- [X] I have made sure this issue exists in the latest version of the plugin
- [X] I have tested with the `minimal.lua` config file above and still get the issue
- [X] I have confirmed that Treesitter is working correctly by running `:checkhealth treesitter`

**Comment by mmirus** [comment]
PS: here are the extra ft highlights I used, if it's useful:

```lua
  ft_highlights = {
    markdown = {
      TSPunctDelimiter = { fg = "${gray}" }, -- e.g., the brackets around links
      TSTextReference = { fg = "${blue}" }, -- e.g., the name part of a link
      TSURI = { fg = "${gray}" }, -- the URL part of a link
      TSLiteral = { fg = "${green}" }, -- e.g., inline code
      TSParameter = { fg = "${fg}" }, -- e.g., function and command args
    },
  },

```

**Comment by olimorris** [comment]
I posted this on Reddit the other day and had no replies. I've trawled the schema's too; comparing the Python and the Markdown ones. Still not sure why some work and others don't

**Comment by mmirus** [comment]
Should we take this upstream to the neovim or treesitter repo? Otherwise, perhaps we just sit in it for a bit and see if an explanation or solution presents itself.

**Comment by olimorris** [comment]
Reading the [commit](https://github.com/neovim/neovim/pull/19931/files) which introduced it, it appears like it *should* just work...

Can you have a read too? Then I think we take it upstream.

**Comment by mmirus** [comment]
It seems like it. And, for example, `:hi @text.literal guifg=Blue` yields:

![image](https://user-images.githubusercontent.com/524994/194782423-ceaa5045-8504-4176-8fa2-a8de1f923b92.png)

But `:hi @text.literal.markdown guifg=Blue` does nothing, even though after running that command running `:hi @text.literal.markdown` prints:

![image](https://user-images.githubusercontent.com/524994/194782449-16b0508c-4a0b-4a41-91ca-20f35d920f43.png)

---

That said, I briefly loaded up a minimal config with treesitter but not this theme, and `:hi @text.literal guifg=Blue` didn't work. It just removed the default `TSLiteral` color. But I only had a minute to try it out. Maybe you can see what results you get with stock neovim + treesitter? If it doesn't work then, it's clearly not local to this theme and we can take it upstream.

**Comment by olimorris** [comment]
With stock Neovim and Treesitter I'm getting the same results:

### minimal.lua
```lua
-- ignore default config and plugins
vim.opt.runtimepath:remove(vim.fn.expand("~/.config/nvim"))
vim.opt.packpath:remove(vim.fn.expand("~/.local/share/nvim/site"))

-- append test directory
local test_dir = "/tmp/treesitter_filetype_test"
vim.opt.runtimepath:append(vim.fn.expand(test_dir))
vim.opt.packpath:append(vim.fn.expand(test_dir))

-- install packer
local install_path = test_dir .. "/pack/packer/start/packer.nvim"
local install_plugins = false

if vim.fn.empty(vim.fn.glob(install_path)) > 0 then
    vim.cmd("!git clone https://github.com/wbthomason/packer.nvim " .. install_path)
    vim.cmd("packadd packer.nvim")
    install_plugins = true
end

local packer = require("packer")

packer.init({
    package_root = test_dir .. "/pack",
    compile_path = test_dir .. "/plugin/packer_compiled.lua",
})

-- install plugins
packer.startup(function(use)
    use("wbthomason/packer.nvim")
    use({ "nvim-treesitter/nvim-treesitter", run = ":TSUpdate" })

    if install_plugins then
        packer.sync()
    end
end)

-- setup treesitter
local ok, treesitter = pcall(require, "nvim-treesitter.configs")
if ok then
    treesitter.setup({
        ensure_installed = "all",
        ignore_install = { "phpdoc" }, -- list of parser which cause issues or crashes
        highlight = { enable = true },
    })
end

vim.cmd("hi clear")
vim.cmd("set termguicolors")

vim.api.nvim_set_hl(0, "Normal", { bg = "#282c34", fg = "#abb2bf" })
vim.api.nvim_set_hl(0, "@text.title.markdown", { fg = "#282c34", bg = "#e06c75" })
vim.api.nvim_set_hl(0, "@punctuation.special.markdown", { fg = "#c678dd", bg = "#98c379" })
vim.api.nvim_set_hl(0, "@text.literal.markdown", { fg = "#abb2bf", bg = "#61afef" })
vim.api.nvim_set_hl(0, "@uri.markdown", { fg = "#282c34", bg = "#e5c07b" })
```

### example.md

```markdown
WORKING with @thing.markdown

"@text.title.markdown"
`vim.api.nvim_set_hl(0, "@text.title.markdown", { fg = "#282c34", bg = "#e06c75" })`
## Here's a title that should be black with a red background

"@punctuation.special.markdown"
`vim.api.nvim_set_hl(0, "@punctuation.special.markdown", { fg = "#c678dd", bg = "#98c379" })`
1. The number 1 should be purple on a green background

NOT working with @thing.markdown

"@text.literal.markdown"
`vim.api.nvim_set_hl(0, "@text.literal.markdown", { fg = "#abb2bf", bg = "#61afef" })`
Here is `inline code that should be grey on a blue background`

"@uri.markdown"
`vim.api.nvim_set_hl(0, "@uri.markdown", { fg = "#282c34", bg = "#e5c07b" })`
The URI in [this markdown link](#some-link) should be black on a yellow background
```

### command

`nvim --clean -u minimal.lua example.md`

If you can confirm the same then I'll take this upstream later today.

**Comment by mmirus** [comment]
Here's what I get when using those files:

![image](https://user-images.githubusercontent.com/524994/194895414-373793d9-14b0-49f9-a35e-21f72f298e84.png)

**Comment by olimorris** [comment]
Yep exactly the same as me. Right, let's pass this up the tree!

**Comment by mmirus** [comment]
Sounds good. Thanks for helping track this down!

**Comment by mmirus** [comment]
Hey @olimorris - would you like me to open an issue over at nvim-treesitter, since the core folks pointed that direction?

**Comment by olimorris** [comment]
I've taken the comment on board with regards to https://github.com/nvim-treesitter/nvim-treesitter/issues/3629. Should have remembered they're two different parsers!

Let me know what you think of 5b20b16be3befda8089f6dbdf2826195f63d753c.

**Comment by mmirus** [comment]
Aaaaah, `_inline` :sweat:. Thanks for tracking that down! The highlights are working now. :partying_face:

**Comment by mmirus** [comment]
@olimorris Related question: do you know if it's possible to define a highlight for compound selectors? (Not sure if "selector" is the right term.)

For example, in vscode, italics in Markdown are purple, including the *s.

If I try that with the current Markdown highlights in the theme, we end up with orange *s because of the `@punctuation.delimiter` highlight:

![image](https://user-images.githubusercontent.com/524994/195428490-59b4ac8c-9d39-4b81-844a-71c769a23d44.png)

If we could define a highlight for `@punctuation.delimiter + @text.emphasis`, we could specifically target asterisks that are part of the italicized text.

![image](https://user-images.githubusercontent.com/524994/195428518-cfa3fcff-c222-48c7-9739-5a5b26d94526.png)

Any ideas?

**Comment by olimorris** [comment]
I've tried playing with this too; it's something I see across all filetypes. No idea about stacking them unfortunately. I've tried soooo many combinations 😆.

I've just asked in that closed Treesitter ticket...cheeky!

**Comment by mmirus** [comment]
Thanks for looking into it! I feel like it's too much to hope for at present, but hopefully will become available at some point. :crossed_fingers:
